props/saber_BC_buttons.h

```diff
@@ -477,738 +1492,1266 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
-    case EVENTID(BUTTON_POWER, EVENT_SECOND_CLICK_LONG, MODE_OFF):
-      if (!mode_volume_) {
-        previous_preset();
-      } else {
-        VolumeDown();
-      }
-      return true;
+  void DoQuote() {
+    if (scroll_presets_ || spam_blast_) return;
+    if (quote_delay_timer_.isTimerRunning()) return;  // prevent overlapping.
+    if (SFX_quote) {
+      sequential_quote_ ? SFX_quote.SelectNext() : SFX_quote.Select(-1);
+      SaberBase::DoEffect(EFFECT_QUOTE, 0);
+    } else {
+      SaberBase::DoForce();
+    }
+  }
 
-// Start or Stop Track
-#if NUM_BUTTONS == 1
-    case EVENTID(BUTTON_POWER, EVENT_FOURTH_SAVED_CLICK_SHORT, MODE_OFF):
-    case EVENTID(BUTTON_POWER, EVENT_FOURTH_SAVED_CLICK_SHORT, MODE_ON):
-#else
-    // 2 or 3 button
-    case EVENTID(BUTTON_POWER, EVENT_SECOND_SAVED_CLICK_SHORT, MODE_ON | BUTTON_AUX):
-    case EVENTID(BUTTON_POWER, EVENT_SECOND_SAVED_CLICK_SHORT, MODE_OFF | BUTTON_AUX):
-#endif
-      if (!mode_volume_) {
-        StartOrStopTrack();
-      }
-      return true;
+  void DoSpamBlast() {
+    if (!spam_blast_) return;
+    SaberBase::DoBlast();
+    last_blast_millis_ = millis();
+  }
 
-// Enter / Exit Volume MENU
-#ifndef NO_VOLUME_MENU
-#if NUM_BUTTONS == 1
-      case EVENTID(BUTTON_NONE, EVENT_CLASH, MODE_OFF | BUTTON_POWER):
-#else
-      // 2 button
-      case EVENTID(BUTTON_AUX, EVENT_FIRST_CLICK_LONG, MODE_OFF):
+  void ToggleSequentialQuote() {
+    if (scroll_presets_ || spam_blast_) return;
+    if (spam_blast_) return;
+    sequential_quote_ = !sequential_quote_;
+    PVLOG_NORMAL << (sequential_quote_ ? "** Quotes play sequentially\n" : "** Quotes play randomly\n");
+    if (SFX_mnum) {
+      sequential_quote_ ? sound_library_v2.SaySequential() : sound_library_.SayRandom();
+    } else {
+      beeper.Beep(0.5, sequential_quote_ ? 3000 : 1000);
+    }
+  }
+
+  void ToggleSpamBlast() {
+    spam_blast_ = !spam_blast_;
+    PVLOG_NORMAL << (spam_blast_ ? "** Entering" : "** Exiting") << " Spam Blast Mode\n";
+    if (!hybrid_font.PlayPolyphonic(&SFX_mzoom)) {
+      spam_blast_ ? BeepEnterFeature() : BeepExitFeature();
+    }
+  }
+
+  void ToggleBattleMode() {
+#ifndef BC_NO_BM
+  battle_mode_ = !battle_mode_;
+    PVLOG_NORMAL << (battle_mode_ ? "** Entering" : "** Exiting") << " Battle Mode\n";
+    if (battle_mode_) {
+      if (!hybrid_font.PlayPolyphonic(&SFX_bmbegin)) {
+        hybrid_font.DoEffect(EFFECT_FORCE, 0);
+      }
+    } else {
+      if (!hybrid_font.PlayPolyphonic(&SFX_bmend)) {
+        beeper.Beep(0.5, 3000);
+      }
+    }
 #endif
-        VolumeMenu();
-        return true;
+  }
+
+#ifdef BC_DUAL_BLADES
+
+  static const BladeSet controlled_blades_;
+
+  bool isMainBladeOn() { return SaberBase::OnBlades()[BC_MAIN_BLADE]; }
+  bool isSecondBladeOn() { return SaberBase::OnBlades()[BC_SECOND_BLADE]; }
+
+  // Get blade location to apply effect to
+  BladeSet GetBladeAboveHorizon() {
+    return accel_.x > 0 ? BC_MAIN_BLADE_SET : BC_SECOND_BLADE_SET;
```

Comment:
Actually, it's probably better to use fusor.down().x instead of accel_

Reply:
or fusor.angle1()

Reply:
fusor.angle1() does work better, thanks.